core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -160,7 +166,7 @@ public void close() throws IOException {
    */
   public synchronized void reset() throws IOException {
     close();
-    init();
+    init(MasterInquireClient.Factory.create());
```

Comment:
Its guaranteed that `create` will give us the correct inquire client?

Reply:
it gets the correct client based on the current `Configuration`. Previously we created it this way on the first line of `init()`, so there shouldn't be any functionality change

Reply:
Can `reset` be called on a `Context` which was created with a passed in `MasterInquireClient`?

Reply:
resetting would replace the `MasterInquireClient` with a new client based on current configuration. I've updated the javadoc to warn about this